### PR TITLE
US888078: Rework Flyway Database Installer usage

### PR DESCRIPTION
**Pull request by kusumaghoshdastidar**
https://internal.almoctane.com/ui/entity-navigation?p=131002*6001&entityType=work_item&id=888078

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -243,10 +243,15 @@
...
+            <dependency>
+                <groupId>com.github.cafapi.util.flywayinstaller</groupId>
+                <artifactId>util-flywayinstaller</artifactId>
+                <version>2.1.0-US888078-SNAPSHOT</version>
+            </dependency>
             <dependency>
                 <groupId>com.github.cafapi.util.flywayinstaller</groupId>
                 <artifactId>util-flywayinstaller-cli</artifactId>
-                <version>2.0.0-184</version>
+                <version>2.1.0-US888078-SNAPSHOT</version>
```
Comment (kusumaghoshdastidar) [inline pom.xml:254]:
```suggestion
                <version>2.1.0-SNAPSHOT</version>
```

File: pom.xml [hunk 2]
```diff
@@ -243,10 +243,15 @@
                 <artifactId>util-ref</artifactId>
                 <version>3.0.0-549</version>
             </dependency>
+            <dependency>
+                <groupId>com.github.cafapi.util.flywayinstaller</groupId>
+                <artifactId>util-flywayinstaller</artifactId>
+                <version>2.1.0-US888078-SNAPSHOT</version>
```
Comment (kusumaghoshdastidar) [inline pom.xml:249]:
```suggestion
                <version>2.1.0-SNAPSHOT</version>
```

File: pom.xml [hunk 3]
```diff
@@ -243,10 +243,15 @@
...
+            <dependency>
+                <groupId>com.github.cafapi.util.flywayinstaller</groupId>
+                <artifactId>util-flywayinstaller</artifactId>
+                <version>3.0.0-US888078-SNAPSHOT</version>
+            </dependency>
             <dependency>
                 <groupId>com.github.cafapi.util.flywayinstaller</groupId>
                 <artifactId>util-flywayinstaller-cli</artifactId>
-                <version>2.0.0-184</version>
+                <version>3.0.0-US888078-SNAPSHOT</version>
```
Comment (kusumaghoshdastidar) [inline pom.xml:254]:
```suggestion
                <version>3.0.0-SNAPSHOT</version>
```

File: pom.xml [hunk 4]
```diff
@@ -243,10 +243,15 @@
                 <artifactId>util-ref</artifactId>
                 <version>3.0.0-549</version>
             </dependency>
+            <dependency>
+                <groupId>com.github.cafapi.util.flywayinstaller</groupId>
+                <artifactId>util-flywayinstaller</artifactId>
+                <version>3.0.0-US888078-SNAPSHOT</version>
```
Comment (kusumaghoshdastidar) [inline pom.xml:249]:
```suggestion
                <version>3.0.0-SNAPSHOT</version>
```

File: job-service-dropwizard/pom.xml [hunk 5]
```diff
@@ -46,6 +46,11 @@
             <groupId>com.github.jobservice</groupId>
             <artifactId>job-service-config</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.github.jobservice</groupId>
+            <artifactId>job-service-db</artifactId>
+            <scope>runtime</scope>
+        </dependency>
```
Comment (dermot-hardy) [inline job-service-dropwizard/pom.xml]: Put this down with the other runtime dependencies.

File: worker-jobtracking-container/src/main/startup.d/setup-database.sh [hunk 6]
```diff
@@ -16,8 +16,9 @@
 #
 
 
-cd /database
-java -jar job-service-db-installer.jar \
+cd /maven
+java -cp "*:classpath" \
```
Comment (dermot-hardy) [inline worker-jobtracking-container/src/main/startup.d/setup-database.sh]: I know it's harmless but why do you keep adding `:classpath` to all these?

```suggestion
java -cp "*" \
```

File: pom.xml [hunk 7]
```diff
@@ -900,6 +918,11 @@
                 <artifactId>jersey-server</artifactId>
                 <version>3.0.12</version>
             </dependency>
+            <dependency>
+                <groupId>com.fasterxml.jackson.dataformat</groupId>
+                <artifactId>jackson-dataformat-toml</artifactId>
+                <version>2.16.1</version>
+            </dependency>
```
Comment (dermot-hardy) [inline pom.xml]: Move this up with the other Jackson dependencies.

File: job-service-db/README.md [hunk 8]
```diff
@@ -11,11 +11,12 @@ directly or you can use the Docker image that is  also available.
...
-With PostgreSQL 11 or later installed and configured, download the [job-service-db-X.X.X-XXX-jar-with-dependencies](https://repo1.
-maven.org/maven2/com/github/jobservice/job-service-db/) jar from Artifactory and then run:
+With PostgreSQL 11 or later installed and configured, download the [job-service-db-X.X.X-XXX](https://repo1.
+maven.org/maven2/com/github/jobservice/job-service-db/) jar from Artifactory into a folder where the rest of its dependency jars are present.
 
-	java -jar /job-service-db-6.0.0-SNAPSHOT-jar-with-dependencies.jar -db.host localhost -db.port 5432 -db.name jobservice -db.pass root 
--db.user postgres
+From the folder with the jars run:
+
+	java -cp "*:classpath" com.github.cafapi.util.flywayinstaller.Application -db.host localhost -db.port 5432 -db.name jobservice -db.pass root -db.user postgres
```
Comment (dermot-hardy) [inline job-service-db/README.md:19]:
```suggestion
	java -cp "*" com.github.cafapi.util.flywayinstaller.Application -db.host localhost -db.port 5432 -db.name jobservice -db.pass root -db.user postgres
```

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**